flink-connector-elasticsearch8/src/test/java/org/apache/flink/connector/elasticsearch/sink/Elasticsearch8TestUtils.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.flink.connector.elasticsearch.sink;
+
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Utility class for Elasticsearch8 tests. */
+public class Elasticsearch8TestUtils {
```

Comment:
Why do we need this class vs using existing `ElasticsearchSinkBaseITCase` ?

Reply:
Good question, it's not required. `ElasticsearchSinkBaseITCase` creates a static ES container while the new test `Elasticsearch8AsyncSinkSecureITCase` needs to create the secure ES container. The new one does not really inherit this base class, but simply refer to those static fields. I think it's a bit clearer to extract those common fields and methods so secure vs. non-secure tests are completely independent.

I don't have strong preference on this and can move it back to the `ElasticsearchSinkBaseITCase` class if we feel it's better to keep them in the base class.

Reply:
> I don't have strong preference on this and can move it back to the ElasticsearchSinkBaseITCase class if we feel it's better to keep them in the base class.

It looks more straightforward to me to be fair (with ElasticsearchSinkBaseITCase), the container creation could be parameterized if needed, thank you

Reply:
Indeed the parameterized tests will reuse the code at the max level and still keep separate cases/parameters independent. I thought of this idea previously but realized the `ES_CONTAINER` field is static (to amortize cost of container setup) and annotated `@Container` for managed lifecycle.

When parameterized, there will be multiple if-else checks depending on the parameter in the base class and child test classes, mainly for ES client and sink builder. This is not a problem per se, and just needs a bit more refactoring.

I'll move the fields / static methods back to the base class for now, and take another look at parameterized tests.

Reply:
@reta I'm working on the parameterized tests, which is purely a test code refactoring:
 - Make base test class parameterized with `secure` parameter. As JUnit 5 has limited support for parameterized tests with inheritance, I used the `ParameterizedTestExtension` introduced in Flink, see [this doc](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit#heading=h.jf7lfvm64da4)
 - Manage the test container lifecycle instead of using the managed annotation `@Testcontainers` and `@Container`
 - Create and use common methods in the base class that concrete test classes can be mostly parameter-agnostic

If you prefer we wait for that change, I can push to this branch after I have a working version. If you agree, I can also create a new PR of the testing code refactoring for future proof (new tests will be easily covered by secure clusters).

Reply:
> If you prefer we wait for that change, I can push to this branch after I have a working version. If you agree, I can also create a new PR of the testing code refactoring for future proof (new tests will be easily covered by secure clusters).

I would agree with you that parameterized tests would very likely make things cleaner (we do have some level of duplication now). If you could pull it off, would be great, I sadly cannot merge, only review, so we still have time till committer comes in.

Reply:
Sounds good, @reta ! Thanks for the review and helpful comments.

Let's see if @snuyanzin has bandwidth to take a look while I'm trying to refactor the tests separately.